go/obsclient/obsclient.go

```diff
@@ -0,0 +1,41 @@
+package obsclient
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/obscuronet/go-obscuro/go/rpcclientlib"
+)
+
+// ObsClient requires an RPC Client and provides access to general Obscuro functionality that doesn't require viewing keys.
+type ObsClient struct {
+	c rpcclientlib.Client
+}
+
+func Dial(rawurl string) (*ObsClient, error) {
```

Comment:
It feels to me that either both ObsClient and AuthObsClient should have a `Dial` method, or neither should. It feels wrong that they're very similar, but have different set up semantics.

Reply:
@joeldudley I've added `DialWithAuth()` because you can't use a clashing name and I think it makes the usage clearer (it requires a priv key).

Reply:
It worries me. So someone can still call ‘Dial’ on the ‘AuthObsClient’, even though it’s meaningless to do so? I think if we don’t want to expose ‘Dial’ on ‘AuthObsClient’ (sounds like we don’t) then we should use another approach to composition here?